src/pages/my_registrations/MyRegistrations.tsx

```diff
@@ -1,67 +1,64 @@
-import { useState } from 'react';
 import { useTranslation } from 'react-i18next';
-import { Box } from '@mui/material';
+import { Box, Typography } from '@mui/material';
 import { Helmet } from 'react-helmet-async';
+import { useQuery } from '@tanstack/react-query';
 import { ListSkeleton } from '../../components/ListSkeleton';
-import { BackgroundDiv } from '../../components/styled/Wrappers';
-import { TabButton } from '../../components/TabButton';
-import { MyRegistrationsResponse, RegistrationStatus } from '../../types/registration.types';
+import { RegistrationStatus } from '../../types/registration.types';
 import { MyRegistrationsList } from './MyRegistrationsList';
-import { useFetch } from '../../utils/hooks/useFetch';
-import { PublicationsApiPath } from '../../api/apiPaths';
+import { fetchRegistrationsByOwner } from '../../api/registrationApi';
 
-enum Tab {
-  Published,
-  Unpublished,
+interface MyRegistrationsProps {
+  selectedPublished: boolean;
+  selectedUnpublished: boolean;
 }
 
-export const MyRegistrations = () => {
+export const MyRegistrations = ({ selectedUnpublished, selectedPublished }: MyRegistrationsProps) => {
   const { t } = useTranslation();
-  const [selectedTab, setSelectedTab] = useState(Tab.Unpublished);
-  const [myRegistrationsResponse, isLoading, refetchRegistrations] = useFetch<MyRegistrationsResponse>({
-    url: PublicationsApiPath.RegistrationsByOwner,
-    errorMessage: t('feedback.error.get_registrations'),
-    withAuthentication: true,
+
+  const registrationsQuery = useQuery({
+    queryKey: ['by-owner'],
+    queryFn: fetchRegistrationsByOwner,
+    meta: { errorMessage: t('feedback.error.search') },
   });
-  const registrations = myRegistrationsResponse?.publications ?? [];
 
-  const unpublishedRegistrations = registrations
-    .filter(({ status }) => status === RegistrationStatus.Draft)
-    .sort((a, b) => new Date(b.createdDate).getTime() - new Date(a.createdDate).getTime());
+  const registrations = registrationsQuery.data?.publications ?? [];
 
-  const publishedRegistrations = registrations
-    .filter(({ status }) => status === RegistrationStatus.Published || status === RegistrationStatus.PublishedMetadata)
-    .sort((a, b) => new Date(b.createdDate).getTime() - new Date(a.createdDate).getTime());
+  const filteredRegistrations = registrations
+    .filter(
+      ({ status }) =>
+        (status === RegistrationStatus.Draft && selectedUnpublished) ||
+        ((status === RegistrationStatus.Published || status === RegistrationStatus.PublishedMetadata) &&
+          selectedPublished)
+    )
+    .sort((a, b) => {
+      if (a.status === RegistrationStatus.Draft || b.status !== RegistrationStatus.Draft) {
```

Comment:
Skulle ikke dette vært AND?
```suggestion
      if (a.status === RegistrationStatus.Draft && b.status !== RegistrationStatus.Draft) {
```

Reply:
(Hvis begge er draft skal de vurderes likt, og vi skulle returnert 0)